lib/logger_json/formatters/ecs.ex

```diff
@@ -0,0 +1,236 @@
+          "file.name" => ~c"/app/logger_json/test/formatters/ecs_test.exs",
+          "file.line" => 18,
+          "function" => "test logs an LogEntry of every level/1"
+        },
+        "message" => "Hello"
+      }
+
+  Example of logging by keywords or by map (Logger.info(%{message: "Hello", foo: :bar, fiz: %{buz: "buz"}})).
+  The keywords or map items are added to the top-level of the log entry:
+
+      %{
+        "@timestamp" => "2024-05-17T16:20:00.000Z",
+        "ecs.version" => "8.11.0",
+        "fiz" => %{"buz" => "buz"},
+        "foo" => "bar",
+        "log.level" => "debug",
+        "log.logger" => "Elixir.LoggerJSON.Formatters.ECSTest",
+        "log.origin" => %{
+          "file.line" => 68,
+          "file.name" => ~c"/app/logger_json/test/formatters/ecs_test.exs",
+          "function" => "test logs an LogEntry with a map payload containing message/1"},
+        "message" => "Hello"
+      }
+
+  Example of logging due to raising an exception (`raise RuntimeError`):
+
+      %{
+        "@timestamp" => "2024-05-17T16:20:00.000Z",
+        "ecs.version" => "8.11.0",
+        "error.message" => "runtime error",
+        "error.stack_trace" => "** (RuntimeError) runtime error\\n    Elixir.LoggerJSON.Formatters.ECSTest.erl:159: anonymous fn/4 in LoggerJSON.Formatters.ECSTest.\\"test logs exceptions\\"/1\\n",
+        "error.type" => "Elixir.RuntimeError",
+        "log.level" => "error",
+        "message" => "runtime error"
+      }
+
+  Note that if you raise an exception that contains an `id` or a `code` property, they will be included in the log entry as `error.id` and `error.code` respectively.
+
+  Example:
+
+      defmodule TestException do
+        defexception [:message, :id, :code]
+      end
+
+      ...
+
+      raise TestException, id: :oops_id, code: 42, message: "oops!"
+
+  results in:
+
+      %{
+        "@timestamp" => "2024-05-17T16:20:00.000Z",
+        "ecs.version" => "8.11.0",
+        "error.code" => 42,
+        "error.id" => "oops_id",
+        "error.message" => "oops!",
+        "error.stack_trace" => "** (LoggerJSON.Formatters.ECSTest.TestException) oops!\n    test/formatters/ecs_test.exs:190: anonymous fn/0 in LoggerJSON.Formatters.ECSTest.\"test logs exceptions with id and code\"/1\n",
+        "error.type" => "Elixir.LoggerJSON.Formatters.ECSTest.TestException",
+        "log.level" => "error",
+        "message" => "oops!"
+      }
+  """
+  import Jason.Helpers, only: [json_map: 1]
+  import LoggerJSON.Formatter.{MapBuilder, DateTime, Message, Metadata, Plug, Encoder}
+
+  @ecs_version "8.11.0"
+
+  @processed_metadata_keys ~w[file line mfa domain error_logger
+                              otel_span_id span_id
+                              otel_trace_id trace_id
+                              conn]a
+
+  @spec format(any(), any()) :: none()
+  def format(%{level: level, meta: meta, msg: msg}, opts) do
+    metadata_keys_or_selector = Keyword.get(opts, :metadata, [])
```

Comment:
We also need redactors support here, this is something that was recently added to master.

Reply:
Yup, I saw that was something you added there, I'll add it in here too!

Reply:
Done!